src/components/ExerciseDetails.tsx

```diff
@@ -46,18 +43,29 @@ function ExerciseDetails({ exercise }: { exercise: Exercise | null }) {
     <Box sx={{ padding: 10 }}>
       <form className="flex-column padding " onSubmit={makeSubmitHandler}>
         <Stack spacing={5}>
-          <h2>{getOperation(exerciseData)} Exercise</h2>
+          <h2>
+            {getOperation(exerciseData)} Exercise {exerciseData.type}
+          </h2>
           <FormControl fullWidth>
-            <InputLabel id="exerciseTypeSelect">Exercise Type</InputLabel>
-            <Select
-              id="exerciseTypeSelect"
-              label="Exercise type"
-              value={exerciseData.type}
-              onChange={makeChangeHandler}
-              required
-            >
-              {showWorkoutOptions(workoutOptions)}
-            </Select>
+            <Autocomplete
+              freeSolo
+              id="exerciseTypeInput"
+              disableClearable
+              options={workoutOptions}
+              renderInput={(params) => (
+                <TextField
+                  // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
You can add a rule to your eslint  rules: {
    'react/jsx-props-no-spreading': 'off',
  }
 if theres no better solution!